warehouse/validations/steps.go

```diff
@@ -15,63 +17,50 @@ func validateStepFunc(_ context.Context, destination *backendconfig.DestinationT
 }
 
 func StepsToValidate(dest *backendconfig.DestinationT) *model.StepsResponse {
-	var (
-		destType = dest.DestinationDefinition.Name
-		steps    []*model.Step
-	)
+	destType := dest.DestinationDefinition.Name
+
+	if destType == warehouseutils.SnowpipeStreaming {
+		dest.Config["useKeyPairAuth"] = true // Since we are currently only supporting key pair auth
```

Comment:
 Can we move this check to [CreateManager](https://github.com/rudderlabs/rudder-server/blob/06e99fa8cc786347fed0dffa102d107ab8e66d1e/warehouse/validations/validate.go#L497) for Snowpipe Streaming.
 Currently we would need to hardcode this because Snowpipe Streaming only supports key pair Authentication?

Reply:
Now that you mention it, I believe the validations package shouldn't be aware of the supported authentication types. That configuration should be provided by the caller. I'll move it to `snowpipestreaming` package